### ISSUE TITLE
[RFE] Allow `Hauler` to pull a specific `platform` or `architecture` of images

### ISSUE THREAD

**Issue by zackbradys** [hauler-dev/hauler#155]
**Is your feature request related to a problem? Please describe.**
* Yes, please see below

**Describe the solution you'd like**
* For `hauler` to have the ability to specify the `platform`/`architecture` when fetching images.
* Possible implementation below (similar functionality as `-k/--key`):

```bash
zackbradys@Zacks-MBP ~ % hauler store add image -h
Add an image to the content store

Usage:
  hauler store add image [flags]

Flags:
  -h, --help         help for image
  -k, --key string   (Optional) Path to the key for digital signature verification
  -p, --platform string   (Optional) Specify the platform/architecture of the image

Global Flags:
      --cache string       Location of where to store cache data (defaults to $XDG_CACHE_DIR/hauler)
  -l, --log-level string    (default "info")
  -s, --store string       Location to create store at (default "store")
```

```bash
apiVersion: content.hauler.cattle.io/v1alpha1
kind: Images
metadata:
  name: hauler-content-images-example
spec:
  images:
    - name: neuvector/scanner:latest
    - name: rgcrprod.azurecr.us/longhornio/longhorn-ui:v1.5.1
      key: carbide-key.pub
      platform: linux/amd64
```

**Describe alternatives you've considered**
* Unfortunately, there is not an easy alternatives to add, remove, delete, etc... specific `platforms` or `architectures`, unless it is natively implemented into `hauler`.

**Additional context**
* Understandably there are outside issues that is preventing an easy implementation of this feature, but it is difficult to effectively use `hauler`, when you are pulling 3 or 4 or even 5 times the amount of images. It may be worth a temporary solution and fix to address this issue
* For example, when using other similar tools, such as `docker` or `skopeo`, the images listed in the published/default `rancher-images.txt` is `32GB`, but when using `hauler` it is `110GB`.

**Comment by amartin120** [comment]
The RGS `cosign` fork now allows for `--platform` on `cosign save` which was a precursor to making this work for Hauler.

https://github.com/rancher-government-carbide/cosign/releases/tag/v2.2.2%2Bcarbide.2

I'll start on incorporating this into Hauler asap.

**Comment by amartin120** [comment]
Available with `v0.4.3`.  https://github.com/rancherfederal/hauler/pull/164

**Comment by zackbradys** [comment]
Closing as complete!
